pkg/controllers/managementuser/rbac/namespace_handler.go

```diff
@@ -53,24 +57,73 @@ type nsLifecycle struct {
 	rq *resourcequota.SyncController
 }
 
-func (n *nsLifecycle) Create(obj *v1.Namespace) (runtime.Object, error) {
+// onChange implements the same functionality as the previous norman-based nsLifecycle:
+// - First ever reconciliation triggers onCreate, after which an annotation is added to mark this event.
+// - Following reconciliations observe this annotation and run a regular update instead
+// - A finalizer is also used to block deletion and trigger cleanup,
+// The original annotation and finalizer keys from norman are used to preserve backwards compatibility
+func (n *nsLifecycle) onChange(_ string, obj *v1.Namespace) (*v1.Namespace, error) {
+	if obj == nil {
+		return nil, nil
+	}
+
+	if obj.DeletionTimestamp != nil {
+		if !slices.Contains(obj.GetFinalizers(), normanLifecycleFinalizer) {
+			// already finalized
+			return obj, nil
+		}
+		return n.onRemove(obj)
+	}
+
+	if obj.Annotations[normanLifecycleAnnotation] != "true" {
+		return n.onCreate(obj)
+	}
+
+	_, err := n.syncNS(obj)
```

Comment:
Since `syncNS` may modify the object passed to it, it should create a deep copy before making any changes. 
I suggest refactoring `syncNS` so that it returns three values: 
- the (possibly modified) object
- a boolean indicating whether a modification was made
-  an error